libcst/_typed_visitor_base.py

```diff
@@ -10,12 +10,10 @@
 if TYPE_CHECKING:
     from libcst._typed_visitor import CSTTypedBaseFunctions  # noqa: F401
 
-T = TypeVar("T")
+F = TypeVar("F", bound=Callable)
 
 
-def mark_no_op(
-    f: Callable[["CSTTypedBaseFunctions", T], None]
```

Comment:
@DragonMinded - this change *does* remove the typing constraint that `mark_no_op` can only decorate this signature of callable, right?

I imagine it's more important to preserve func signature than type the decorated function.

Reply:
Honestly, there's no safety concern in accidentally decorating a class or a function that we don't care about. I would rather have type safety of decorated functions, especially since `mark_no_op` is an implementation detail of LibCST core. This is fine.